client/src/lib/data.ts

```diff
@@ -0,0 +1,70 @@
+export type Movie = {
+  title: string;
+  summary: string;
+  link: string;
+  rating: number;
+  movieId?: number;
+};
+
+export async function getMovies(): Promise<Movie[] | null> {
+  const url = "http://localhost:3000/api/movies";
+  try {

```

Comment:
You shouldn't use try/catch here. Instead, it should be used up in the component. These non-component functions should just throw, and the component will catch the error and display the error to the user.